core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -1072,6 +1073,11 @@ public SystemSessionProperties(
                 durationProperty(CLOSE_IDLE_WRITERS_TRIGGER_DURATION,
                         "The duration after which the writer operator tries to close the idle writers",
                         new Duration(5, SECONDS),
+                        true),
+                booleanProperty(
+                        PUSH_FILTER_INTO_VALUES,
```

Comment:
why do we need a session property for this?

if this is just a kill switch (it's hidden), let's have a code comment and maybe mark it deprecated

Reply:
The property should be visible. I added it because some users define huge datasets through VALUES (although this is a misuse) and in such case the optimization would be inefficient. I will refactor the property to limit the size of VALUES to be optimized (see other comment).